csm_web/scheduler/admin.py

```diff
@@ -49,273 +88,311 @@ def has_delete_permission(self, request, obj=None):
-
-    def get_fields(self, request, obj=None):
-        fields = ["name", "course", "role", "section", "user"]
-        if request.user.is_superuser:
-            fields.insert(2, "leader")
-        if obj and obj.role == "ST":
-            fields.append("get_attendances")
-            fields.append("active")
-        return tuple(fields)
-
-    def get_queryset(self, request):
-        qs = super().get_queryset(request)
-        if request.user.is_superuser:
-            return qs
-        return qs.filter(
-            course__in=[
-                p.course
-                for p in request.user.profile_set.filter(
-                    active=True, role=Profile.COORDINATOR
-                )
-            ]
-        )
-
-    def save_model(self, request, obj, form, change):
-        if obj.role == Profile.STUDENT:
-            obj.leader = obj.section.leader
-        super().save_model(request, obj, form, change)
-
-    def deactivate_profiles(self, request, queryset):
-        queryset.update(active=False)
-
-    deactivate_profiles.short_description = "Mark selected profiles as inactive"
-
-    def activate_profiles(self, request, queryset):
-        queryset.update(active=True)
-
-    activate_profiles.short_description = "Mark selected profiles as active"
-
-    def get_attendances(self, obj):
-        attendance_links = []
-        for attendance in obj.attendance_set.all():
-            admin_url = reverse(
-                "admin:scheduler_attendance_change", args=(attendance.id,)
-            )
-            attendance_links.append(
-                format_html(
-                    '<a style="display: block" href="{}">Week of {}: {}</a>',
-                    admin_url,
-                    attendance.week_start,
-                    attendance.presence if attendance.presence else "--",
-                )
-            )
-        attendance_links.sort()
-        return format_html("".join(attendance_links))
-
-    get_attendances.short_description = "Attendances"
-
-
 @admin.register(Spacetime)
 class SpacetimeAdmin(CoordAdmin):
-    fields = ("location", "day_of_week", "start_time", "duration")
-    list_display = ("location", "day_of_week", "start_time")
-    list_filter = ("location", "day_of_week")
-    search_fields = ("location", "day_of_week")
+    fields = ("_location", "_day_of_week", "_start_time", "_duration")
+    list_display = ("_location", "_day_of_week", "_start_time")
+    list_filter = ("_location", "_day_of_week")
+    search_fields = ("_location", "_day_of_week")
 
     def has_module_permission(self, request, obj=None):
         return request.user.is_superuser  # should only be editable through section
 
     def has_delete_permission(self, request, obj=None):
         return False  # will break 1to1 invariants if deleted
 
-    def get_queryset(self, request):
-        return Spacetime.objects.all()
-
 
 @admin.register(Course)
-class CourseAdmin(admin.ModelAdmin):
+class CourseAdmin(CoordAdmin):
```

Comment:
Again with fixing the `get_queryset` to prefetch

Reply:
I don't think it's possible here, since all the references here are reverse foreign keys

Reply:
Have to use `prefetch_related`, which is the sibling to `select_related`: https://docs.djangoproject.com/en/2.2/ref/models/querysets/#prefetch-related

Reply:
Neat. Also added this to MentorAdmin and SectionAdmin for student set lookup.